### PR TITLE
ARROW-14558 [R] clarify OOP system wording in the Arrow vignette

### PR DESCRIPTION
**Pull request by dragosmg**

### CODE REVIEW

**Comment by dragosmg** [comment]
@nealrichardson @thisisnic @paleolimbot @jonkeane what do you think? 🤔

File: r/vignettes/arrow.Rmd [hunk 1]
```diff
@@ -216,7 +216,7 @@ Note that the `attributes()` stored in `$metadata$r` are only understood by R. I
 
 ## Class structure and package conventions
 
-C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented as `R6` reference classes, most of which are exported from the namespace.
+C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented using the `R6` object-oriented programming system, and most of them are exported from the namespace.
```
Comment (nealrichardson) [inline r/vignettes/arrow.Rmd:219]:
```suggestion
C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented as `R6` classes, most of which are exported from the namespace.
```

Comment (nealrichardson) [reply r/vignettes/arrow.Rmd:219]: Could also link to the R6 package website

Comment (dragosmg) [reply r/vignettes/arrow.Rmd]: Should we also link to Hadley's R6 (sub)chapter in _Advanced R_? Or would anyone interested find that as it's not really a hidden resource?

**Comment by thisisnic** [comment]
Hooray for more clarity!

**Comment by jonkeane** [comment]
This looks good, thank you!

**Pull request closed without merging**